src/ApparentHorizons/SpherePackIterator.cpp

```diff
@@ -0,0 +1,137 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+//
+// SpherePack memory layout
+//
+// Given a S2 with Nth x Nph grid points, the spectral coefficients
+// are stored as follows:
+//
+//                         Mmax+1
+//     -----------------------------------------------------
+//     | (0,0)       x          x         x
+//     | (1,0)     (1,1)        x         x         ARRAY
+//     | (2,0)     (2,1)      (2,2)       x          "a"
+//     |   :         :          :         x
+//     | (Mmax,  0) ...        (Mmax,   Mmax)
+//     | (Mmax+1,0) ...        (Mmax+1, Mmax)
+//     |    :                        :
+//     | (Lmax, 0)             (Lmax,   Mmax)
+//     -----------------------------------------------------
+//     |    x        x          x         x
+//     |    x      (1,1)        x         x
+//     |    x      (2,1)      (2,2)       x         ARRAY
+//     |    :        :          :         x          "b"
+//     |    x    (Mmax,  1)..  (Mmax,   Mmax)
+//     |    x    (Mmax+1,1)..  (Mmax+1, Mmax)
+//     |    :         :              :
+//     |    x    (Lmax,  1)..  (Lmax,   Mmax)
+//     -----------------------------------------------------
+//
+// This layout has 2(Ntheta) rows and Mmax+1 columns,
+// the numbers across a *row* are contiguous.
+// 'x' stands for a number that is not referenced by SpherePack.
+// (Note that there is one complete unreferenced row in the middle)
+//
+// In the above, Lmax is equal to Nth-1
+//               Mmax is equal to min(Lmax,Nphi/2) [integer division]
+//
+//               so the full size of each array is (Lmax+1)*(Mmax+1)
+//
+// Note that Spherepack is ok with arbitrary values of Nth and Nphi, which
+// it takes as input, and computes Lmax=Nth-1 and Mmax=min(Lmax,Nphi/2).
+// However, some combinations of Nth and Nphi have strange properties:
+// - The maximum M that is AT LEAST PARTIALLY represented by (Nth,Nph)
+//   is std::min(Nth-1,Nphi/2). This is called Mmax here. But an arbitrary
+//   (Nth,Nph) does not necessarily fully represent all m's up to M_max,
+//   because sin(M_max phi) might be zero at all collocation points, and
+//   therefore sin(M_max phi) might not be representable on the grid.
+// - The largest m that is fully represented by (Nth,Nph) is
+//   M_maxrepresented = std::min(Nth-1,(Nphi-1)/2).
+// - Therefore, if Nphi is odd,  M_max = M_maxrepresented,
+//              if Nphi is even, M_max = M_maxrepresented+1.
+// - If L and M_maxrepresented are specified, then setting
+//      Nth = L+1
+//      Nph = 2*M_maxrepresented+1
+//   will ensure that M_max = M_maxrepresented.
```

Comment:
I have some suggestions for this comment (mostly intending to improve the clarity of some points) that I will post later...